### DISCUSSION TITLE
Viewer  with PyQt Backend  -Matplotlib Backend example

### DISCUSSION THREAD

**Discussion by rdejtiar** [mozman/ezdxf#874] [Q&A]
Hi everyone 

Using this great library I want to  implement a viewer with PyQt backend like the existing with Ezdxf view , plus the functionality of maintaining  previous zoom and position when the file is reloaded  trough a reload button .

Like the example with matplotlib attached.

[ezdxf_matplotlib_viewr_B .txt](https://github.com/mozman/ezdxf/files/11412643/ezdxf_matplotlib_viewr_B.txt).

To call the viewer from the main script would be nice too

Could anybody give me a hand with this?

Thanks in advance

**Comment by mbway** [comment]
does this PR add the functionality you were looking for? https://github.com/mozman/ezdxf/pull/882

for 'running the viewer from the main script' you might also be interested in this change https://github.com/mozman/ezdxf/pull/880

**Comment by rdejtiar** [comment]
Hi Matt
Thank you for your answer.
I could not test the change.
what i did is to download the modified qtviewer.py and try to check with
-ezdxf view - through the CLI

(base) xx:~$ ezdxf view
using Qt binding: PyQt5
Traceback (most recent call last):
  File "/home/ricardo/mambaforge/bin/ezdxf", line 11, in <module>
    sys.exit(main())
  File
"/home/ricardo/mambaforge/lib/python3.10/site-packages/ezdxf/__main__.py",
line 98, in main
    run(args)
  File
"/home/ricardo/mambaforge/lib/python3.10/site-packages/ezdxf/commands.py",
line 452, in run
    from ezdxf.addons.drawing.qtviewer import CadViewer
ImportError: cannot import name 'CadViewer' from
'ezdxf.addons.drawing.qtviewer'
(/home/ricardo/mambaforge/lib/python3.10/site-packages/ezdxf/addons/drawing/qtviewer.py)

The problems are (for me at least) :
class CADViewer(qw.QMainWindow):
    def __init__(self, cad: Optional[CADWidget] = None):

"CADViewer" was formerly "CadViewer"
What I could check on cad_viewer.py .   I was using cad_viewer.py for my
own tests.
The next line also is not recognized as was previously  v =
CADViewer(config=config) on cad_viewer . I could not see how ezdxf view
works . Itś beyond my expertise.

Thanks for your help and sorry if iḿ messing up .

Ricardo

**Reply by mbway** [reply]
the best way to use the new functionality before the next ezdxf release is to uninstall ezdxf from your virtualenvironment, clone the ezdxf repository and run
```
cd <ezdxf repo>
pip install .
```
After the next release you can uninstall your editable copy and install from pypi or conda forge again

**Comment by rdejtiar** [comment]
Thanks Matt
It works on demand (with Reload) and automatically (watch marked)
It also keeps the view when ticked.
The only issue I see , and don't  know if related with this addition  is
that text and attributes are not shown, only the container, even when they
are shown on the element properties on the sidebar.
I'ḿ using Qt binding: PyQt5.
Best Regards
Ricardo

**Reply by mbway** [reply]
Are these elements visible before clicking refresh?

**Comment by rdejtiar** [comment]
No. They are not visible at the first file load

**Reply by mbway** [reply]
then the issue is likely unrelated. Please open a separate issue for that, ideally with an example
